Fix flaky search by reason test

### Summary

The search by reason test is flaky and sometimes fails.

```python
    def test_searches_by_reason(self):
        agency = factories.AgencyFactory()
        url = reverse("nc:agency-api-searches-by-type", args=[agency.pk])
    
        type_code, type_label = SEARCH_TYPE_CHOICES[2]
    
        # Create the following racial data for 2015: 1 black
        p1 = factories.PersonFactory(race="B", ethnicity="N", stop__agency=agency, stop__year=2015)
        factories.SearchFactory(person=p1, stop=p1.stop, type=type_code)
    
        # Create the following racial data for 2016: 1 native american, 3 hispanic
        p2 = factories.PersonFactory(race="W", ethnicity="H", stop__agency=agency, stop__year=2016)
        factories.SearchFactory(person=p2, stop=p2.stop, type=type_code)
        p3 = factories.PersonFactory(race="B", ethnicity="H", stop__agency=agency, stop__year=2016)
        factories.SearchFactory(person=p3, stop=p3.stop, type=type_code)
        p4 = factories.PersonFactory(race="B", ethnicity="H", stop__agency=agency, stop__year=2016)
        factories.SearchFactory(person=p4, stop=p4.stop, type=type_code)
        p5 = factories.PersonFactory(race="I", ethnicity="N", stop__agency=agency, stop__year=2016)
        factories.SearchFactory(person=p5, stop=p5.stop, type=type_code)
        StopSummary.refresh()
    
        response = self.client.get(url, format="json")
        response_data = response.json()
        self.assertEqual(response.status_code, status.HTTP_200_OK)
        # Two years = two items
        self.assertEqual(len(response_data), 2)
    
        self.assertEqual(response_data[0]["year"], 2015)
        self.assertEqual(response_data[0]["black"], 1)
        self.assertEqual(response_data[0]["search_type"], type_label)
        self.assertEqual(response_data[1]["year"], 2016)
>       self.assertEqual(response_data[1]["hispanic"], 3)
E       AssertionError: 2 != 3
```

## AC

- [ ] Test is deterministic and passes

--- a/nc/tests/factories.py
+++ b/nc/tests/factories.py
@@ -3,6 +3,8 @@ import datetime
 import factory
 import factory.fuzzy
 
+from django.utils.timezone import localtime
+
 from nc import models
 
 
@@ -46,6 +48,8 @@ class StopFactory(factory.django.DjangoModelFactory):
         if extracted:
             day = 1 if self.date.month == 2 else self.date.day
             self.date = self.date.replace(year=extracted, day=day)
+            if localtime(self.date).year != extracted:
+                self.date += datetime.timedelta(1)
 
 
 class SearchFactory(factory.django.DjangoModelFactory):

